exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporter.java

```diff
@@ -35,51 +40,66 @@ public final class HttpExporter<T extends Marshaler> {
 
   private final String type;
   private final HttpSender httpSender;
-  private final ExporterMetrics exporterMetrics;
+  private final ExporterMetricsAdapter exporterMetrics;
 
   public HttpExporter(
-      String exporterName,
-      String type,
+      String legacyExporterName,
+      ExporterMetrics.Signal type,
+      ComponentId componentId,
       HttpSender httpSender,
       Supplier<MeterProvider> meterProviderSupplier,
-      boolean exportAsJson) {
-    this.type = type;
+      HealthMetricLevel healthMetricLevel,
+      boolean exportAsJson,
+      Attributes healthMetricAttributes) {
+    this.type = type.toString();
     this.httpSender = httpSender;
+    // TODO: extract server.address and server.port here
     this.exporterMetrics =
-        exportAsJson
-            ? ExporterMetrics.createHttpJson(exporterName, type, meterProviderSupplier)
-            : ExporterMetrics.createHttpProtobuf(exporterName, type, meterProviderSupplier);
+        new ExporterMetricsAdapter(
+            healthMetricLevel,
+            meterProviderSupplier,
+            type,
+            componentId,
+            healthMetricAttributes,
+            legacyExporterName,
+            exportAsJson ? "http-json" : "http");
   }
 
   public CompletableResultCode export(T exportRequest, int numItems) {
     if (isShutdown.get()) {
       return CompletableResultCode.ofFailure();
     }
 
-    exporterMetrics.addSeen(numItems);
+    ExporterMetricsAdapter.Recording metricRecording =
+        exporterMetrics.startRecordingExport(numItems);
 
     CompletableResultCode result = new CompletableResultCode();
 
     httpSender.send(
         exportRequest,
         exportRequest.getBinarySerializedSize(),
-        httpResponse -> onResponse(result, numItems, httpResponse),
-        throwable -> onError(result, numItems, throwable));
+        httpResponse -> onResponse(result, metricRecording, httpResponse),
+        throwable -> onError(result, metricRecording, throwable));
 
     return result;
   }
 
   private void onResponse(
-      CompletableResultCode result, int numItems, HttpSender.Response httpResponse) {
+      CompletableResultCode result,
+      ExporterMetricsAdapter.Recording metricRecording,
+      HttpSender.Response httpResponse) {
     int statusCode = httpResponse.statusCode();
 
+    Attributes requestAttributes =
+        Attributes.builder().put(SemConvAttributes.HTTP_RESPONSE_STATUS_CODE, statusCode).build();
```

Comment:
#nit
```suggestion
Attributes requestAttributes = Attributes.of(SemConvAttributes.HTTP_RESPONSE_STATUS_CODE, (long) statusCode);
```

Also, is there no good way to abstract building these attributes into the `metricRecording` API?